client/apps/edit2/components/content/sections/images/components/controls.jsx

```diff
@@ -63,10 +78,21 @@ export class ImagesControls extends Component {
   }
 
   onNewImage = (image) => {
-    const { section } = this.props
-    const newImages = clone(section.get('images')).concat(image)
+    const {
+      editSection,
+      isHero,
+      section,
+      onChangeHeroAction,
+      onChangeSectionAction
+    } = this.props
 
-    section.set('images', newImages)
+    if (isHero) {
+      const newImages = clone(section.images).concat(image)
+      onChangeHeroAction('images', newImages)
+    } else {
+      const newImages = clone(editSection.images).concat(image)
+      onChangeSectionAction('images', newImages)
+    }
```

Comment:
👍this is the sort of thing that could have been written cleverly but straightforward and clear is always better